src/components/CollectionDialog.js

```diff
@@ -163,6 +155,13 @@ export class CollectionDialog extends Component {
     const { filter } = this.state;
 
     if (error) return null;
+    // If this component is optimistically rendering ahead of the window its in
+    // force a re-render so that it is placed correctly. The right thing here is
+    // to maybe pass a ref.
+    if (!document.querySelector(`#${containerId} #${windowId}`)) {
+      this.forceUpdate();
```

Comment:
An odd part of this for sure.. but felt like this was sufficient until we start passing around `Window` refs.